be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -102,6 +115,409 @@ Status BetaRowsetWriter::add_block(const vectorized::Block* block) {
     return _add_block(block, &_segment_writer);
 }
 
+vectorized::VMergeIterator* BetaRowsetWriter::_get_segcompaction_reader(
```

Comment:
I recommend create a new file such as seg_compaction.cpp to handle segment compaction and logic in beta_rowset_writer could be much lightweight. In segment compaction, call init、do_impl、finish interface so that this code would be much clear to review

Reply:
Thanks for your suggestion. It seems to need large modification on the patch and re-test. So I'd rather do it in further patches (maybe when add multi-pass segcompaction). If this commit itself is good, I suggest to merge it as it is -- a step by step methodology.